zenoh-plugin-ros2dds/src/route_service_cli.rs

```diff
@@ -351,24 +352,46 @@ fn route_dds_request_to_zenoh(
         return;
     }
 
-    let zbuf: ZBuf = sample.into();
-    let slice = zbuf.to_zslice();
+    let z_bytes: ZBytes = sample.into();
+    let slice = Cow::from(z_bytes);
     let dds_req_buf = slice.as_ref();
+
     let is_little_endian =
         is_cdr_little_endian(dds_req_buf).expect("Shouldn't happen: sample.len >= 20");
+
     let request_id = CddsRequestHeader::from_slice(
         dds_req_buf[4..20]
             .try_into()
             .expect("Shouldn't happen: sample.len >= 20"),
         is_little_endian,
     );
 
-    // route request buffer stripped from request_id (client_id + sequence_number)
-    let mut zenoh_req_buf = ZBuf::empty();
     // copy CDR Header
-    zenoh_req_buf.push_zslice(slice.subslice(0, 4).unwrap());
+    let cdr_header = match dds_req_buf.get(0..4) {
```

Comment:
Couldn't we just use dds_req_buf[0..4]?

Reply:
using `dds_req_buf[0..4]` is the same thing, however it is unsafe. 
So if we are going to remove all the unwraps and expects, slicing directly into an array is a sort of silent unwrap that we could introduce if the buffer isn't long enough.
Of course we do have the check at the top about buffer length, but its possible that a programmer changes some assumption about the buffer length and in the future it causes a silent failure, explicitly handling cases where this can happen, while more verbose, does save us from potential future failures.